torch/_export/utils.py

```diff
@@ -89,6 +89,56 @@ def _collect_and_set_constant_attrs(
     return constant_attrs
 
 
+def _register_constants_as_buffers(
+    mod: torch.fx.GraphModule, state_dict, non_persistent_buffers
+):
+    # TODO some annoying circular dependency issue
+    from torch.export.unflatten import _assign_attr, _AttrKind
+
+    temp_registered_constants = set()
+
+    for node in mod.graph.nodes:
+        if node.op == "get_attr":
+            target = torch.fx.graph_module._get_attr(mod, node.target)
+            if isinstance(target, torch.Tensor):
+                # Make sure we also check if the original buffer is
+                # non persistent as well.
+                if (node.target not in state_dict) and (
+                    node.target not in non_persistent_buffers
+                ):
+                    torch.fx.graph_module._del_attr(mod, node.target)
```

Comment:
the new private graph_module method feels a bit sus. There are existing methods in `fx.graph` for replacing usages of one node with another, are they not enough?

Reply:
This is removing an attribute instead of manipulating nodes. There are existing methods in that module for getting attributes, so precedent.

Reply:
Yep i think we do need this because when we assign a buffer to graph module, it complains that there is already existing attribute. I could create a fresh new graph and initialize things properly but it seems better to just delete the attr in the first place.